src/external/smartsheet.ts

```diff
@@ -0,0 +1,135 @@
+import createHttpError from 'http-errors';
+import { StatusCodes } from 'http-status-codes';
+import fetch from 'node-fetch';
+
+import { smartsheetId, smartsheetToken } from '../config/env';
+import config from '../config/project';
+import { IUserOuput } from '../db/models/User';
+
+const parseRoles = (roles: string[]) =>
+    roles.map((role) => config.roleOptions.find((option) => option.value === role)?.label || role).join(', ');
+
+export const handleNewsletterUpdate = async (user: IUserOuput): Promise<any> => {
+    switch (user.newsletter_subscription_status) {
+        case 'subscribed':
+            await subscribeNewsletter(user);
+            break;
+        case 'unsubscribed':
+            await unsubscribeNewsletter(user.newsletter_email);
+            break;
+        default:
+            break;
+    }
+};
+
+export const subscribeNewsletter = async (user: IUserOuput): Promise<any> => {
+    const url = `https://api.smartsheet.com/2.0/sheets/${smartsheetId}/rows`;
+    const headers = {
+        Authorization: `Bearer ${smartsheetToken}`,
+        'Content-Type': 'application/json',
+    };
+
+    const { newsletter_email, first_name, last_name, roles, affiliation } = user;
+
+    if (!newsletter_email) {
+        throw createHttpError(
+            StatusCodes.BAD_REQUEST,
+            'Some required fields are missing to complete newsletter subscription',
+        );
+    }
+
+    const { results } = await fetchSubscription(newsletter_email);
+
+    if (results.length > 0) return;
+
+    const body = [
+        {
+            toTop: true,
+            cells: [
+                {
+                    columnId: 2634539247921028,
```

Comment:
are columnId newsletter specific ? meaning if we want to subscribe to a newsletter for another project, would these ids be the same ?